From cc4b44179a793eb5b6764d42fba770bfae233138 Mon Sep 17 00:00:00 2001
From: Alex Gonzalez <alexg@balena.io>
Date: Fri, 2 Jul 2021 20:11:11 +0200
Subject: [PATCH] nwl_dsi-imx: Delays before and after clock
 enabling/disabling.

Introduce delays before clock setups to guarantee parent PLLs clock
stability and after clock setup to gurantee the clocks are stable before
being used.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>
---
 drivers/gpu/drm/imx/nwl_dsi-imx.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/imx/nwl_dsi-imx.c b/drivers/gpu/drm/imx/nwl_dsi-imx.c
index 658619936745..07f16594e8a1 100644
--- a/drivers/gpu/drm/imx/nwl_dsi-imx.c
+++ b/drivers/gpu/drm/imx/nwl_dsi-imx.c
@@ -502,8 +502,12 @@ static void imx_nwl_dsi_enable(struct imx_mipi_dsi *dsi)
 
 	request_bus_freq(BUS_FREQ_HIGH);
 
+	msleep(100);
+
 	imx_nwl_dsi_set_clocks(dsi, true);
 
+	msleep(20);
+
 	ret = devtype->poweron(dsi);
 	if (ret < 0) {
 		DRM_DEV_ERROR(dev, "Failed to power on DSI (%d)\n", ret);
@@ -528,8 +532,12 @@ static void imx_nwl_dsi_disable(struct imx_mipi_dsi *dsi)
 	if (!dsi->no_clk_reset)
 		devtype->poweroff(dsi);
 
+	msleep(10);
+
 	imx_nwl_dsi_set_clocks(dsi, false);
 
+	msleep(8);
+
 	release_bus_freq(BUS_FREQ_HIGH);
 
 	dsi->enabled = false;
